platform/ios/src/MGLScaleControlView.m

```diff
@@ -0,0 +1,48 @@
+#import "MGLScaleControlView.h"
+
+@interface MGLScaleControlView()
+@property (nonatomic, assign) double zoomLevel;
+@property (nonatomic, assign) CLLocationDistance metersPerPoint;
+@property (nonatomic) NSArray *labels;
+@property (nonatomic) NSArray *bars;
+@property (nonatomic) UILabel *debugLabel;
+@end
+
+@implementation MGLScaleControlView
+
+- (void)updateWithZoomLevel:(double)zoomLevel metersPerPoint:(CLLocationDistance)metersPerPoint {
+    self.zoomLevel = zoomLevel;
+    self.metersPerPoint = metersPerPoint;
+}
+
+- (void)setZoomLevel:(double)zoomLevel {
+    _zoomLevel = zoomLevel;
+    [self.debugLabel setText:[NSString stringWithFormat:@"%0.2f", zoomLevel]];
+    [self.debugLabel sizeToFit];
+    [self setNeedsLayout];
+}
+
+- (CGSize)intrinsicContentSize {
+    return CGSizeMake(100, 10);
+}
+
+- (UILabel *)debugLabel {
+    if (!_debugLabel) {
+        _debugLabel = [[UILabel alloc] init];
+        _debugLabel.font = [UIFont systemFontOfSize:8];
+        [self addSubview:_debugLabel];
```

Comment:
I prefer to use -drawInRect: when I want instant rendering instead of default animation between layouts otherwise I prefer -layoutSubviews. Working directly with the views also improves readability but I haven't really decided yet. The logic should be the same so we can switch back if we don't get the expected behavior without much fuss.